hawc/apps/lit/serializers.py

```diff
@@ -196,3 +196,24 @@ def bulk_create_tags(self):
             models.ReferenceTags.objects.bulk_create(new_tags)
 
             models.Reference.delete_cache(assessment_id)
+
+
+class ReferenceSerializer(serializers.ModelSerializer):
+    tags = serializers.ListField(write_only=True)
+
+    @transaction.atomic()
+    def update(self, instance, validated_data):
+
+        # updates the reference tags
+        if "tags" in validated_data:
+            instance.tags.set(validated_data.pop("tags"))
```

Comment:
Just realized it could be the case the ids are not from this assessment.  This needs to be a little more involved. It probably belongs in the `validate` method:

```python
tag_ids = validated_data.pop("tags")
tags = ReferenceFilterTag.get_assessment_qs(self.instance.assessment_id).filter(id__in=tag_ids)
if tags.count() != len(tag_ids):
    raise serializer.ValidationError('...')
validated_data[tags] = tags

```